tests/suites/test_suite_ssl.function

```diff
@@ -3669,6 +3670,64 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_TEST_HOOKS:MBEDTLS_SSL_PROTO_TLS1_3 */
+void psa_hkdf_extract( int alg, char *hex_ikm_string,
+                       char *hex_salt_string, char *hex_prk_string )
+{
+    unsigned char *ikm = NULL;
+    unsigned char *salt = NULL;
+    unsigned char *prk = NULL;
+    unsigned char *output_prk = NULL;
+    size_t ikm_len, salt_len, prk_len, output_prk_len;
+
+    output_prk_len = PSA_HASH_LENGTH( alg );
+    output_prk = mbedtls_calloc( 1, output_prk_len );
+
+    ikm = mbedtls_test_unhexify_alloc( hex_ikm_string, &ikm_len );
+    salt = mbedtls_test_unhexify_alloc( hex_salt_string, &salt_len );
+    prk = mbedtls_test_unhexify_alloc( hex_prk_string, &prk_len );
+
+    PSA_ASSERT( psa_crypto_init() );
+    PSA_ASSERT( mbedtls_psa_hkdf_extract( alg, salt, salt_len,
+                                          ikm, ikm_len, output_prk ) );
+
+    ASSERT_COMPARE( output_prk, output_prk_len, prk, prk_len );
+
+exit:
+    mbedtls_free(ikm);
+    mbedtls_free(salt);
+    mbedtls_free(prk);
+    mbedtls_free(output_prk);
+
+    PSA_DONE( );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_TEST_HOOKS:MBEDTLS_SSL_PROTO_TLS1_3 */
+void psa_hkdf_extract_ret( int hash_len, int ret )
+{
+    int output_ret;
+    unsigned char *salt = NULL;
+    unsigned char *ikm = NULL;
+    unsigned char *prk = NULL;
+    size_t salt_len, ikm_len;
+
+    prk = mbedtls_calloc( PSA_MAC_MAX_SIZE, 1 );
+    salt_len = hash_len;
```

Comment:
This differs from the `mbedtls_` version, I'm not saying that's wrong but can you explain the difference?

Reply:
After a second look it does not seems good. This function tests the length of the hash is invalid, but there is no way to specify it with this parameter set. I change it to test the `alg` parameter is invalid.